#assigns associated gene coordinates, from RH genome's gff3 file, to a large table of DEGs generated by DESeq2, specifically for the RH datasets

setwd('D:/DE')

DEG <- read.table("D:/DE/DEG/tuber_russetburbank_DEG.csv", sep = ",", header = TRUE)
View(DEG)
# DEG <- DEG[which(!is.na(DEG$padj)),]
# DEG <- DEG[which(DEG$pvalue < 1),]
# View(DEG)

functional_annotation_file <- read.csv("atl.pm.functional_annotation.txt", sep = "\t", header = FALSE)
View(functional_annotation_file)
colnames(functional_annotation_file) <- c('Gene', 'Annotation')
name <- function(x){
  name <- substr(x, 1, 21)
  return(name)
}
functional_annotation_names <- as.data.frame(functional_annotation_file$Gene)
# View(functional_annotation_names)
functiona_annotation_names_column <- as.data.frame(apply(functional_annotation_names, MARGIN = 1, FUN = name))
# View(functional_annotation_names_column)
functional_annotation_table <- cbind(functional_annotation_names_column, functional_annotation_file$Annotation)
colnames(functional_annotation_table) <- c('Gene', 'Annotation')
# View(functional_annotation_table)

get_annotation <- function(x){
  if (x %in% functional_annotation_table$Gene){
    ann <- toString(functional_annotation_table[which(x == functional_annotation_table$Gene),]$Annotation[1])
  }else{
    ann <- NA
  }
  return(ann)
}

res_names <- as.data.frame(DEG$X)
# View(res_names)
annotation_column <- as.data.frame(apply(res_names, MARGIN = 1, FUN = get_annotation))
# View(annotation_column)

res <- cbind(DEG, annotation_column)
# View(res)
res_annotated <- res
# View(res_annotated)


gff <-  as.data.frame(read.delim("D:/DE/functional_annotation/DEG.hc.pm.locus_assign.gff3", header = F, comment.char = "#"))
colnames(gff) <-  c("seqid", "source", "type", "start", "end", "score", "strand", "phase", "attributes")
# View(gff)
gff_subset <- gff[which(gff$type == 'gene'),]

chr <- function(x){
  if (startsWith(x, 'ID=Soltu.Atl.S')){
    chr <- substr(x, 14, 20)
  }else if (startsWith(x, 'ID=Soltu.Atl.')){
    chr <- substr(x, 14, 15)
  }else if (startsWith(x, 'Parent=Soltu.Atl.')){
    chr <- substr(x, 18, 19)
  }
  return(chr)
}
chromosome_string <- function(x){
  return('chr')
}
colon_string <- function(x){
  return(':')
}
dash_string <- function(x){
  return('-')
}
name <- function(x){
  name <- sub(".*=", "", x)
  return(name)
}

attribute <- function(x){
  if (startsWith(x, 'ID=')){
    if (startsWith(x, 'ID=Soltu.Alt.S')){
      attribute <- substr(x, 4, 27)
    }else{
      attribute <- substr(x, 14, 21)
    }
  }else if(startsWith(x, 'Parent=')){
    attribute <- substr(x, 8, 30)
  }
  return(attribute)
}

name_column <- as.data.frame(name(gff_subset$attributes))
attributes <- as.data.frame(gff_subset$attributes)
start_column <- as.data.frame(gff_subset$start)
end_column <- as.data.frame(gff_subset$end)
attribute_column <- as.data.frame(apply(attributes, MARGIN = 1, FUN = attribute))
chromosome_column <- as.data.frame(apply(attributes, MARGIN = 1, FUN = chr))
chr_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = chromosome_string))
colon_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = colon_string))
dash_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = dash_string))

gff_subset <- cbind(gff_subset, name_column)
gff_subset <- cbind(gff_subset, chr_column)
gff_subset <- cbind(gff_subset, chromosome_column)
gff_subset <- cbind(gff_subset, colon_column)
gff_subset <- cbind(gff_subset, start_column)
gff_subset <- cbind(gff_subset, dash_column)
gff_subset <- cbind(gff_subset, end_column)
# View(gff_subset)

colnames(gff_subset) <- c('seqid', 'source', 'type', 'start1', 'end1', 'score', 'strand', 'phase', 'attributes', 'attributes1', 'chr', 'chromosome', 'colon', 'start', 'dash', 'end')
gff_subset$coordinates <- with(gff_subset, paste0(gff_subset$chr, gff_subset$chromosome, gff_subset$colon, gff_subset$start, gff_subset$dash, gff_subset$end))
# View(gff_subset)
gff_subset <- gff_subset[c('seqid', 'source', 'type', 'start1', 'end1', 'score', 'strand', 'phase', 'attributes1', 'coordinates')]
colnames(gff_subset) <- c('seqid', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes', 'coordinates')
# View(gff_subset)
res_final <- cbind(name_column, gff_subset$coordinates)
# View(res_final)
write.csv(as.data.frame(res_final), file = 'tuber_russetburbank_DEG_coordinates.csv')

coordinates <- function(x){
  coordinates <- res_final[which(x == res_final$attributes),]$coordinates[1]
  return(coordinates)
}

coordinate_column <- as.data.frame(apply(res_names, MARGIN = 1, FUN = coordinates))
View(coordinate_column)
DEG <- cbind(DEG, coordinate_column)
# Different colnames and reordering for edgeR and ballgown produced DEG tables
# colnames(DEG) <- c('X', 'geneNames', 'transcriptNames', 'transcriptIDs', 'feature', 'id', 'fc', 'pval', 'qval', 'coordinates')
# DEG <- DEG[c('transcriptNames', 'transcriptIDs', 'feature', 'id', 'fc', 'pval', 'qval', 'coordinates')]
# colnames(DEG) <- c('ID', 'baseMean', 'log2foldchange', 'lfcSE', 'pvalue', 'padj', 'annotation', 'coordinates')
# colnames(DEG) <- c('GeneID', 'logFC', 'logCPM', 'PValue', 'FDR', 'Annotation', 'Coordinates')
colnames(DEG) <- c('ID', 'baseMean', 'log2foldchange', 'lfcSE', 'pvalue', 'padj', 'coordinates')
# View(DEG)
write.csv(DEG, file = 'tuber_russetburbank_DEG_coordinates.csv', row.names = FALSE)


