#assigns associated gene coordinates, from RH genome's gff3 file, to a large table of DEGs generated by DESeq2, specifically for the RH datasets

setwd('D:/DE')

output_filename <- 'RHatl_DE_TP1_annotated.csv'
DEG <- read.table(file = 'RHatl_TP1_DE.csv', sep = ",", header = TRUE)
# View(RH)
gff <-  as.data.frame(read.delim("RH89-039-16_potato_gene_models.v3.gff3", header = F, comment.char = "#"))
colnames(gff) <-  c("seqid", "source", "type", "start", "end", "score", "strand", "phase", "attributes")
# View(gff)
res_names <- as.data.frame(DEG$X)
# View(res_names)
res_annotated <- DEG
# View(res_annotated)

gff_subset <- gff[which(gff$type == 'gene'),]
gff_subset <- gff_subset[c('attributes')]
# View(gff_subset)

chromosome <- function(x){
  chromosome<- substr(x, 7, 8)
}
chromosome_string <- function(x){
  return('chr')
}
colon_string <- function(x){
  return(':')
}
dash_string <- function(x){
  return('-')
}
name_subset <- function(x){
  name_subset <- sub(".*name_subset=", "", x)
  name_subset <- substr(name_subset, 1, 14)
  return(name_subset)
}
name_subset_column <- as.data.frame(name_subset(gff_subset$attributes))
attributes <- as.data.frame(gff_subset$attributes)
start_column <- as.data.frame(gff[which(gff$type == 'gene'),])$start
end_column <- as.data.frame(gff[which(gff$type == 'gene'),])$end
chromosome_column <- as.data.frame(apply(attributes, MARGIN = 1, FUN = chromosome))
chromosome_string_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = chromosome_string))
colon_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = colon_string))
dash_column <- as.data.frame(apply(gff_subset, MARGIN = 1, FUN = dash_string))

gff_subset <- cbind(name_subset_column, chromosome_string_column)
gff_subset <- cbind(gff_subset, chromosome_column)
gff_subset <- cbind(gff_subset, colon_column)
gff_subset <- cbind(gff_subset, start_column)
gff_subset <- cbind(gff_subset, dash_column)
gff_subset <- cbind(gff_subset, end_column)
colnames(gff_subset) <- c('attributes', 'chr', 'chromosome', 'colon', 'start', 'dash', 'end')
# View(gff_subset)

gff_subset$coordinates <- with(gff_subset, paste0(gff_subset$chr, gff_subset$chromosome, gff_subset$colon, gff_subset$start, gff_subset$dash, gff_subset$end))
# View(gff_subset)
res_final <- gff_subset[c('attributes', 'coordinates')]
# View(res_final)
colnames(res_final) <- c('attribute', 'coordinate')
# View(res_final)

get_coordinates <- function(x){
  coordinates <- res_final[which(x == res_final$attribute),]$coordinate[1]
  return(coordinates)
}
coordinate_column <- as.data.frame(apply(res_names, MARGIN = 1, FUN = get_coordinates))
# View(coordinate_column)

DEG <- cbind(res_annotated, coordinate_column)
colnames(DEG) <- c('ID', 'baseMean', 'log2foldchange', 'lfcSE', 'pvalue', 'padj', 'GO', 'IPR', 'SwissProt', 'TAIR', 'coordinates')
# View(DEG)
write.csv(DEG, file = output_filename, row.names = FALSE)

